lib/shopping_cart.rb

```diff
@@ -0,0 +1,51 @@
+class ShoppingCart
+    attr_reader :name, :capacity , :products, :details
+    def initialize(name, capacity, products = [])
+        @name = name
+        @capacity = capacity.to_i
+        @products = products
+        @details = {name: @name,  capacity: @capacity.to_i, }
```

Comment:
We want to steer away from formatting in our initialize method.. It can get a bit convoluted if you get into this habit. Logic looks good, but I'd actually pull this into a method instead of in initialize.